backend/services/inventory/tests/tests/test_api_internal.py

```diff
@@ -12,74 +12,98 @@
 #    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #    See the License for the specific language governing permissions and
 #    limitations under the License.
+
 import os
-from common import (
-    internal_client,
-    management_client,
-    mongo,
-    clean_db,
-    inventory_attributes,
-)
-import bravado
-import pytest
 
+import openapi_client as oas
+import openapi_client.exceptions as api_exceptions
 
-class TestInternalApiTenantCreate:
-    def test_create_ok(self, internal_client, clean_db):
+from client import make_authenticated_client
 
-        _, r = internal_client.create_tenant("foobar")
-        assert r.status_code == 201
+
+class TestInternalApiTenantCreate:
+    def test_create_ok(self, clean_db):
+        internal_client = oas.InventoryInternalApi()
+        rsp = internal_client.create_tenant_with_http_info(
+            oas.TenantNew(tenant_id="foobar")
+        )
+        assert rsp.status_code == 201
 
         assert "inventory-foobar" in clean_db.list_database_names()
         assert "migration_info" in clean_db["inventory-foobar"].list_collection_names()
 
-    def test_create_twice(self, internal_client, clean_db):
-
-        _, r = internal_client.create_tenant("foobar")
-        assert r.status_code == 201
+    def test_create_twice(self, clean_db):
+        internal_client = oas.InventoryInternalApi()
+        rsp = internal_client.create_tenant_with_http_info(
+            oas.TenantNew(tenant_id="foobar")
+        )
+        assert rsp.status_code == 201
 
         # creating once more should not fail
-        _, r = internal_client.create_tenant("foobar")
-        assert r.status_code == 201
+        rsp = internal_client.create_tenant_with_http_info(
+            oas.TenantNew(tenant_id="foobar")
+        )
+        assert rsp.status_code == 201
 
-    def test_create_empty(self, internal_client):
+    def test_create_empty(self):
         try:
-            _, r = internal_client.create_tenant("")
-        except bravado.exception.HTTPError as e:
-            assert e.response.status_code == 400
+            internal_client = oas.InventoryInternalApi()
+            rsp = internal_client.create_tenant_with_http_info(
+                oas.TenantNew(tenant_id="foobar")
+            )
+        except api_exceptions.BadRequestException:
+            pass
```

Comment:
are you sure it also throws? apart from the returning the response?

Reply:
This assertion is indeed weak as it pass if the request succeeds. As mentioned above, the `with_http_info` also calls `response_deserialize` which asserts a successful status code.